### PR TITLE
IAM Users: Add/remove keys tests

### PR DESCRIPTION
**Pull request by swapnil-khandare**
Signed-off-by: Swapnil Khandare <swapnil.khandare@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [x] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

**Comment by swapnil-khandare** [comment]
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37774
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37775
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37776
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37777
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37778
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37779
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37780
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37781
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37782
PASSED tests/csm/rest/test_iam_users.py::TestIamUserRGW::test_37783

**Review by aditipande96 — APPROVED** [review]

File: tests/csm/rest/test_iam_users.py [hunk 1]
```diff
@@ -968,3 +971,450 @@ def test_37020(self):
...
+            uid = payload["tenant"] + "$" + uid
+            uids.append(uid)
+            self.created_iam_users.add(uid)
+            get_resp = self.csm_obj.get_iam_user(user=uid)
+            assert_utils.assert_true(get_resp.status_code == HTTPStatus.OK, "Get IAM user failed")
+            access_key = get_resp.json()["keys"][0]['access_key']
+            access_keys.append(access_key)
+        self.log.info("Adding key to user")
+        add_resp = self.csm_obj.add_key_to_iam_user(uid=uids[0], access_key=access_keys[1])
+        assert_utils.assert_true(add_resp.status_code == HTTPStatus.CONFLICT, "Status Failed")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_iam_users.py:1194]: Error message check is missing.

File: tests/csm/rest/test_iam_users.py [hunk 2]
```diff
@@ -968,3 +971,450 @@ def test_37020(self):
...
+    def test_37777(self):
+        """
+        Create key request with existing access key of another user
+        """
+        test_case_name = cortxlogging.get_frame()
+        self.log.info("##### Test started -  %s #####", test_case_name)
+        self.log.info("[START] Creating IAM user")
+        uids = []
+        access_keys = []
+        for user in range(2):
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_iam_users.py:1175]: can we accept 2 from the test config?

File: tests/csm/rest/test_iam_users.py [hunk 3]
```diff
@@ -968,3 +971,450 @@ def test_37020(self):
...
+        resp = self.csm_obj.create_iam_user_rgw(payload)
+        self.log.info("Verify Response : %s", resp)
+        assert_utils.assert_true(resp.status_code == HTTPStatus.CREATED, "IAM user creation failed")
+        uid = payload["tenant"] + "$" + uid
+        self.created_iam_users.add(uid)
+        access_key = resp.json()["keys"][0]['access_key']
+        add_resp = self.csm_obj.add_key_to_iam_user(uid=uid, access_key=access_key + "123",
+                                                    login_as="csm_user_monitor")
+        assert_utils.assert_true(add_resp.status_code == HTTPStatus.FORBIDDEN,
+                                 "Add key status failed")
```
Comment (kachhwahadivya) [inline tests/csm/rest/test_iam_users.py:1413]: error message check missing

File: tests/csm/rest/test_iam_users.py (hunk 2 above)
Comment (swapnil-khandare) [reply tests/csm/rest/test_iam_users.py:1175]: This is fixed, not going to change. So, will keep as it is.

File: tests/csm/rest/test_iam_users.py (hunk 1 above)
Comment (swapnil-khandare) [reply tests/csm/rest/test_iam_users.py:1194]: Added

File: tests/csm/rest/test_iam_users.py (hunk 3 above)
Comment (swapnil-khandare) [reply tests/csm/rest/test_iam_users.py:1421]: Added

**Review by kachhwahadivya — APPROVED** [review]

**Review by sonalk0209 — APPROVED** [review]
Approving for merge

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**